src/parser/jobs/ast/modules/Lightspeed.tsx

```diff
@@ -1,199 +0,0 @@
-import {t} from '@lingui/macro'
-import {Trans} from '@lingui/react'
-import {DataLink} from 'components/ui/DbLink'
-import {RotationTable} from 'components/ui/RotationTable'
-import {Event, Events} from 'event'
-import {Analyser} from 'parser/core/Analyser'
-import {EventHook} from 'parser/core/Dispatcher'
-import {filter} from 'parser/core/filter'
-import {dependency} from 'parser/core/Injectable'
-import CastTime from 'parser/core/modules/CastTime'
-import {Data} from 'parser/core/modules/Data'
-import {Timeline} from 'parser/core/modules/Timeline'
-import React, {Fragment} from 'react'
-import {Icon, Message} from 'semantic-ui-react'
-import DISPLAY_ORDER from './DISPLAY_ORDER'
-
-const BASE_GCDS_PER_WINDOW = 6
-const LIGHTSPEED_REDUCTION = -2500
-
-interface LightspeedWindow {
-	start: number
-	end?: number
-
-	rotation: Array<Events['action']>
-	gcdCount: number
-	mpSavings: number
-	trailingGcdEvent?: Events['action']
-
-	deathTruncated: boolean
-}
-
-// in this module we only want to track Lightspeed windows opened by
-// the character selected for analysis. windows that clip into
-// AST Lightspeed will be marked.
-// Used DNC Technicalities as basis for this module. Rewritten from previous module for consistency purposes
-export class Lightspeed extends Analyser {
-	static override handle = 'Lightspeed'
-	static override title = t('ast.lightspeed.title')`Lightspeed`
-	static override displayOrder = DISPLAY_ORDER.LIGHTSPEED
-
-	@dependency private timeline!: Timeline
-	@dependency private data!: Data
-	@dependency private castTime!: CastTime
-
-	private history: LightspeedWindow[] = []
-	private currentWindow: LightspeedWindow | undefined = undefined
-	private castHook?: EventHook<Events['action']>
-	private castTimeIndex: number | null = null
-
-	override initialise() {
-		const lightspeedFilter = filter<Event>().status(this.data.statuses.LIGHTSPEED.id)
-
-		this.addEventHook(lightspeedFilter.type('statusApply')
-			.target(this.parser.actor.id), this.tryOpenWindow)
-		this.addEventHook(lightspeedFilter.type('statusRemove')
-			.target(this.parser.actor.id), this.tryCloseWindow)
-	}
-
-	private tryOpenWindow(event: Events['statusApply']) {
-		if (this.currentWindow === undefined) {
-			this.currentWindow = {
-				start: event.timestamp,
-				rotation: [],
-				gcdCount: 0,
-				mpSavings: 0,
-
-				deathTruncated: false,
-
-			}
-			this.castHook = this.addEventHook(
-				filter<Event>()
-					.source(this.parser.actor.id)
-					.type('action'),
-				this.onCast,
-			)
-			this.castTimeIndex = this.castTime.setTimeAdjustment('all', LIGHTSPEED_REDUCTION)
```

Comment:
This code (and the matching code on lines 94 and 95 for tryCloseWindow) is critical for marking spells as instant casts during a Lightspeed window, and needs to be ported to _somewhere_.  There's a visibility/maintainability argument for keeping a separate Lightspeed module to do this castTime modification, or I think having the Lightspeed handling part of the Divination module handle it would be fine as well (though my preference is the more discoverable separate Lightspeed module, even though it wouldn't be suggesting / recommending anything).

Without this code, we're dinging multi-weaves in lightspeed (see http://localhost:3000/fflogs/A6TWmwzX9xkQbtva/117/84)

Reply:
very good point! I see what you mean
I'll add in a lightspeed module and reference it in divination as suggested

Reply:
added lightspeed section
lemme know if it's sufficient
thank you ♥